staging/src/k8s.io/client-go/metadata/metadata_test.go

```diff
@@ -0,0 +1,181 @@
+)
+
+func TestClient(t *testing.T) {
+	gvr := schema.GroupVersionResource{Group: "group", Version: "v1", Resource: "resource"}
+
+	writeJSON := func(t *testing.T, w http.ResponseWriter, obj runtime.Object) {
+		data, err := json.Marshal(obj)
+		if err != nil {
+			t.Fatal(err)
+		}
+		w.Header().Set("Content-Type", "application/json")
+		if _, err := w.Write(data); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	testCases := []struct {
+		name    string
+		handler func(t *testing.T, w http.ResponseWriter, req *http.Request)
+		want    func(t *testing.T, client *Client)
+	}{
+		{
+			name: "GET is able to convert a JSON object to PartialObjectMetadata",
+			handler: func(t *testing.T, w http.ResponseWriter, req *http.Request) {
+				if req.Header.Get("Accept") != "application/vnd.kubernetes.protobuf;as=PartialObjectMetadata;g=meta.k8s.io;v=v1,application/json;as=PartialObjectMetadata;g=meta.k8s.io;v=v1,application/json" {
+					t.Fatal(req.Header.Get("Accept"))
+				}
+				if req.Method != "GET" && req.URL.String() != "/apis/group/v1/namespaces/ns/resource/name" {
+					t.Fatal(req.URL.String())
+				}
+				writeJSON(t, w, &corev1.Pod{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Pod",
+						APIVersion: "v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "name",
+						Namespace: "ns",
+					},
+				})
+			},
+			want: func(t *testing.T, client *Client) {
+				obj, err := client.Resource(gvr).Namespace("ns").Get("name", metav1.GetOptions{})
+				if err != nil {
+					t.Fatal(err)
+				}
+				if obj.GetName() != "name" {
+					t.Fatal(obj)
+				}
+			},
+		},
+
+		{
+			name: "LIST is able to convert a JSON object to PartialObjectMetadata",
+			handler: func(t *testing.T, w http.ResponseWriter, req *http.Request) {
+				if req.Header.Get("Accept") != "application/vnd.kubernetes.protobuf;as=PartialObjectMetadataList;g=meta.k8s.io;v=v1,application/json;as=PartialObjectMetadataList;g=meta.k8s.io;v=v1,application/json" {
+					t.Fatal(req.Header.Get("Accept"))
+				}
+				if req.Method != "GET" && req.URL.String() != "/apis/group/v1/namespaces/ns/resource" {
+					t.Fatal(req.URL.String())
+				}
+				writeJSON(t, w, &corev1.PodList{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "PodList",
+						APIVersion: "v1",
+					},
+					ListMeta: metav1.ListMeta{
+						ResourceVersion: "253",
+					},
+					Items: []corev1.Pod{
+						{
+							TypeMeta: metav1.TypeMeta{
+								Kind:       "Pod",
+								APIVersion: "v1",
+							},
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "name",
+								Namespace: "ns",
+							},
+						},
+					},
+				})
+			},
+			want: func(t *testing.T, client *Client) {
+				objs, err := client.Resource(gvr).Namespace("ns").List(metav1.ListOptions{})
+				if err != nil {
+					t.Fatal(err)
+				}
+				if objs.GetResourceVersion() != "253" {
+					t.Fatal(objs)
+				}
+				expect := []metav1.PartialObjectMetadata{
+					{
+						TypeMeta: metav1.TypeMeta{
+							Kind:       "Pod",
```

Comment:
I didn't expect this kind/apiVersion in the resulting PartialObjectMetadata object

Reply:
I'll double check whether I had a reason to want the originating type.  If not, clearing it seems reasonable (although any code that uses partial object metadata from this client and then expects type meta to be set to anything is...)

Reply:
cleared it

Reply:
yeah... leaving the original (non-partialobjectmetadata) kind/apiVersion in it seems like behavior someone would accidentally depend on, then get bitten by when running against a newer server